src/Symfony/Component/HttpKernel/Debug/DebugRequestStack.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\HttpKernel\Debug;
+
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\RequestStack;
+use Symfony\Component\HttpKernel\Profiler\Profiler;
+
+/**
+ * A stack able to deal with virtual requests.
+ *
+ * Pushing virtual requests to the stack enables the {@see Profiler} collect.
+ * They are obfuscated by default to prevent side effects.
+ *
+ * @internal
+ */
+final class DebugRequestStack extends RequestStack
```

Comment:
shouldn't we deal with mix of nested virtual/non-virtual requests?
push should also be added, and we should throw a logic exception when pushing a virtual request without passing ignoreVirtual = false

Reply:
`VirtualRequestStack`?

Reply:
Renamed.